src/main/java/http_server/ResponseMaker.java

```diff
@@ -4,31 +4,86 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 
+import static http_server.StatusCodes.NO_CONTENT;
+import static http_server.StatusCodes.NOT_FOUND;
+import static http_server.StatusCodes.OK;
+
 public class ResponseMaker {
 
-    public String returnFileContents(String file) {
+    public String statusResponse(String file) {
+        String statusCode = checkIfResourceIsAvailable(file);
+
+        if (statusCode.equals(OK.getStatusCode())) {
+            return buildStatusLine(OK);
+        } else if (statusCode.equals(NOT_FOUND.getStatusCode())) {
+            return buildStatusLine(NOT_FOUND);
+        } else if (statusCode.equals(NO_CONTENT.getStatusCode())) {
+            return buildStatusLine(NO_CONTENT);
+        } else {
+            return "Unhandled file type";
+        }
+    }
+
+    public String checkIfResourceIsAvailable(String resource) {
+        if (requestIsToHomePage(resource)) {
+            return OK.getStatusCode();
+        } else {
+            if (returnResourceContents(resource) != null) {
+                return OK.getStatusCode();
+            } else {
+                return NOT_FOUND.getStatusCode();
+            }
+        }
+    }
+
+    private boolean requestIsToHomePage(String resource) {
+        return resource.equals("/");
+    }
+
+    public String returnResourceContents(String resource) {
         byte[] encodedContents;
         String contents = null;
-        try {
-            String filePath = String.format("public/%s", file);
-            encodedContents = Files.readAllBytes(Paths.get(filePath));
-            contents = new String(encodedContents, "US-ASCII");
-        } catch (IOException e) {
-            e.printStackTrace();
+
+        if (requestIsToHomePage(resource)) {
+            return "";
+        } else {
+            String filePath = String.format("public/%s", resource);
+            try {
+                encodedContents = Files.readAllBytes(Paths.get(filePath));
+                contents = new String(encodedContents, "US-ASCII");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return contents;
         }
-        return contents;
     }
 
-    public String buildWholeResponse(String fileContents) {
+    public String buildWholeResponse(String resourceContents, String resource, String typeOfRequest) {
         String response = "";
+        if (isHeadRequest(typeOfRequest)) {
+            return returnNoMessageBody(resourceContents, resource, response);
+        } else {
+            return returnMessageBody(resourceContents, resource, response);
+        }
+    }
+
+    private String returnMessageBody(String fileContents, String file, String response) {
+        response = response + statusResponse(file) + "\n\n";
+        return response + fileContents;
+    }
+
+    private String returnNoMessageBody(String fileContents, String file, String response) {
+        response = response + statusResponse(file) + "\n";
```

Comment:
similar comment here about using the passed in response as a temp variable (or accumulator, however you want to look at it).